gammapy/scripts/analysis.py

```diff
@@ -51,6 +60,124 @@ def settings(self):
         """Configuration settings for the analysis session."""
         return self.config.settings
 
+    def get_observations(self):
+        """Fetch observations from the data store according to criteria defined in the configuration."""
+        self.config.validate()
+
+        datastore_path = make_path(self.settings["observations"]["datastore"])
+        if datastore_path.is_file():
+            self.datastore = DataStore().from_file(datastore_path)
+        elif datastore_path.is_dir():
+            self.datastore = DataStore().from_dir(datastore_path)
+        else:
+            log.error("Datastore {} not found.".format(datastore_path))
+            return False
+
+        ids = set()
```

Comment:
This reformatting of obs select config is super long. Can it be avoided by directly using the existing
https://docs.gammapy.org/0.13/api/gammapy.data.ObservationTable.html#gammapy.data.ObservationTable.select_observations
or by modifying that method or adding a new one on obs-table (maybe with a TODO comment or reminder issue to unify the two methods again in the future)?

Note that Analysis should only dispatch to existing functionality throughout Gammapy, it shouldn't re-implement new functionality. The code you're adding would need review and work, if really a lot of code is needed obs selection API & implementation & tests should be a separate PR ideally.

Reply:
Yes, I agree this all or most of this should be handled at `ObservationTable.select_observations`. I have reduced a little bit the amount of code, I will create a PR to improve `ObservationTable.select_observations` and get rid of all these tens of lines.

Reply:
Thanks!